File: c/392/main.c
#include <stdbool.h>
#include <string.h>
#include <stdio.h>

bool isSubsequence(char* s, char* t) {
    int i = 0;
    int j = 0;
    int sLen = strlen(s);
    int tLen = strlen(t);
    while (i < sLen && j < tLen) {
        if (s[i] == t[j]) {
            i++;
        }
        j++;
    }
    return i >= sLen;
}

int main() {
    char* s = "abc";
    char* t = "ahbgdc";
    printf("%d\n", isSubsequence(s, t));
    s = "axc";
    t = "ahbgdc";
    printf("%d\n", isSubsequence(s, t));
}

File: go/392/main.go
package main

import "fmt"

func main() {
	s := "abc"
	t := "ahbgdc"
	fmt.Println(isSubsequence(s, t))
	s = "axc"
	t = "ahbgdc"
	fmt.Println(isSubsequence(s, t))
	s = "b"
	t = "c"
	fmt.Println(isSubsequence(s, t))
}

func isSubsequence(s string, t string) bool {
	i := 0
	j := 0
	for i < len(s) && j < len(t) {
		if s[i] == t[j] {
			i++
			j++
		} else {
			j++
		}
	}
	return i >= len(s)
}
